homeassistant/bootstrap.py

```diff
@@ -73,6 +73,63 @@ def _handle_requirements(hass: core.HomeAssistant, component,
     return True
 
 
+def get_validated_component_config(hass: core.HomeAssistant, domain: str,
+                                   config) -> bool:
+    """Validate config for a domain, including all platforms."""
+    component = loader.get_component(domain)
+    missing_deps = [dep for dep in getattr(component, 'DEPENDENCIES', [])
```

Comment:
This will not help in being here for your scripts, as validate script will not be able to do this.